src/http-api.js

```diff
@@ -389,6 +389,10 @@ module.exports.MatrixHttpApi.prototype = {
         prefix,
         accessToken,
     ) {
+        if (!this.opts.idBaseUrl) {
+            throw new Error("No ID Server Base URL set");
```

Comment:
Maybe let's be slightly more formal and say "identity server"?

```suggestion
            throw new Error("No Identity Server base URL set");
```

I also noticed this much more verbose error style:

https://github.com/matrix-org/matrix-js-sdk/blob/17724fc8d35f75d1a7a751b62da6db98798ef548/src/client.js#L2438-L2442

I guess it's just a historic thing we used to do for SDK errors a while ago?

Reply:
Mmm, that looks like it's basically pretending to be a response from the server when it isn't, so not sure about copying that one.